From 1ee117044907657bf5a05fd0c65c7ef17571f5fb Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:40:17 -0400
Subject: [alsa] virtuoso: add Xonar Essence STX II support

Message-id: <140972823573-409-patchset-jkysela@redhat.com>
Patchwork-id: 91105
O-Subject: [RHEL 7.1 PATCH 408/442] ALSA: virtuoso: add Xonar Essence STX II support
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Just add the PCI ID for the STX II.  It appears to work the same as the
STX, except for the addition of the not-yet-supported daughterboard.

Tested-by: Mario <fugazzi99@gmail.com>
Tested-by: corubba <corubba@gmx.de>
Cc: <stable@vger.kernel.org>
Signed-off-by: Clemens Ladisch <clemens@ladisch.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: f42bb22243d2ae264d721b055f836059fe35321f)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/sound/alsa/ALSA-Configuration.txt b/Documentation/sound/alsa/ALSA-Configuration.txt
index 95731a0..8f08b2a 100644
--- a/Documentation/sound/alsa/ALSA-Configuration.txt
+++ b/Documentation/sound/alsa/ALSA-Configuration.txt
@@ -2026,8 +2026,8 @@ Prior to version 0.9.0rc4 options had a 'snd_' prefix. This was removed.
   -------------------
 
     Module for sound cards based on the Asus AV66/AV100/AV200 chips,
-    i.e., Xonar D1, DX, D2, D2X, DS, Essence ST (Deluxe), Essence STX,
-    HDAV1.3 (Deluxe), and HDAV1.3 Slim.
+    i.e., Xonar D1, DX, D2, D2X, DS, DSX, Essence ST (Deluxe),
+    Essence STX (II), HDAV1.3 (Deluxe), and HDAV1.3 Slim.
 
     This module supports autoprobe and multiple cards.
 
diff --git a/sound/pci/Kconfig b/sound/pci/Kconfig
index 3bd0bb1..69b03aa 100644
--- a/sound/pci/Kconfig
+++ b/sound/pci/Kconfig
@@ -855,8 +855,8 @@ config SND_VIRTUOSO
  select SND_JACK if INPUT=y || INPUT=SND
  help
    Say Y here to include support for sound cards based on the
-   Asus AV66/AV100/AV200 chips, i.e., Xonar D1, DX, D2, D2X, DS,
-   Essence ST (Deluxe), and Essence STX.
+   Asus AV66/AV100/AV200 chips, i.e., Xonar D1, DX, D2, D2X, DS, DSX,
+   Essence ST (Deluxe), and Essence STX (II).
    Support for the HDAV1.3 (Deluxe) and HDAV1.3 Slim is experimental;
    for the Xense, missing.
 
diff --git a/sound/pci/oxygen/virtuoso.c b/sound/pci/oxygen/virtuoso.c
index 64b9fda..dbbbacf 100644
--- a/sound/pci/oxygen/virtuoso.c
+++ b/sound/pci/oxygen/virtuoso.c
@@ -53,6 +53,7 @@ static DEFINE_PCI_DEVICE_TABLE(xonar_ids) = {
  { OXYGEN_PCI_SUBID(0x1043, 0x835e) },
  { OXYGEN_PCI_SUBID(0x1043, 0x838e) },
  { OXYGEN_PCI_SUBID(0x1043, 0x8522) },
+ { OXYGEN_PCI_SUBID(0x1043, 0x85f4) },
  { OXYGEN_PCI_SUBID_BROKEN_EEPROM },
  { }
 };
diff --git a/sound/pci/oxygen/xonar_pcm179x.c b/sound/pci/oxygen/xonar_pcm179x.c
index c8c7f2c..e026059 100644
--- a/sound/pci/oxygen/xonar_pcm179x.c
+++ b/sound/pci/oxygen/xonar_pcm179x.c
@@ -100,8 +100,8 @@
  */
 
 /*
- * Xonar Essence ST (Deluxe)/STX
- * -----------------------------
+ * Xonar Essence ST (Deluxe)/STX (II)
+ * ----------------------------------
  *
  * CMI8788:
  *
@@ -1138,6 +1138,14 @@ int get_xonar_pcm179x_model(struct oxygen *chip,
   chip->model.resume = xonar_stx_resume;
   chip->model.set_dac_params = set_pcm1796_params;
   break;
+ case 0x85f4:
+  chip->model = model_xonar_st;
+  /* TODO: daughterboard support */
+  chip->model.shortname = "Xonar STX II";
+  chip->model.init = xonar_stx_init;
+  chip->model.resume = xonar_stx_resume;
+  chip->model.set_dac_params = set_pcm1796_params;
+  break;
  default:
   return -EINVAL;
  }
-- 
1.7.1